File: 2_stack_queue_heap/이준우/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    q = deque()
    pq = deque(sorted(priorities, reverse = True))
    
    for i in range(0, len(priorities)):
        # 현재 인덱스, 우선순위
        q.append((i, priorities[i]))
        
    while(q):
        if pq[0] == q[0][1]:
            answer += 1
            if q[0][0] == location:
                return answer
            q.popleft()
            pq.popleft()
        else:
            q.append(q.popleft())
    
    return answer

File: 2_stack_queue_heap/김민주/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        Queue <Integer> q = new LinkedList<>();
        Queue <Integer> qnum = new LinkedList<>();
        PriorityQueue <Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        int ln = priorities.length;
        boolean flag = false;
        
        for (int i=0; i<ln; i++){
            q.offer(priorities[i]);
            qnum.offer(i);
            pq.offer(priorities[i]);
        }
        
        
        while (!flag){
            int now = q.poll();
            int nownum = qnum.poll();
            int biggest = pq.peek();
            
            if (now == biggest){
                answer +=1;
                pq.poll();
                // return
                if (location == nownum){
                    flag = true;
                    break;
                }
            }else {
                q.offer(now);
                qnum.offer(nownum);
            }
            
        }
    
        return answer;
    }
}